src/util/ConfigManager/ConfigManager.h

```diff
@@ -186,10 +194,78 @@ class ConfigManager {
   */
   std::string printConfigurationDoc(bool printCurrentJsonConfiguration) const;
 
+  /*
+  @brief Add a function to the configuration manager for verifying, that the
+  values of the given configuration options are valid. Will always be called
+  after parsing.
+
+  @tparam ValidatorParameterTypes The types of the values in the
+  `validatorParameter`. Can be left up to type deduction.
+
+  @param validatorFunction Checks, if the values of the given configuration
+  options are valid. Should return true, if they are valid.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param validatorParameter Proxies for the configuration options, whos values
+  will be passed to the validator function as function arguments. Will keep the
+  same order.
+  */
+  template <typename... ValidatorParameterTypes>
+  void addValidator(
+      Validator<ValidatorParameterTypes...> auto validatorFunction,
+      std::string_view errorMessage,
+      ConfigOptionProxy<ValidatorParameterTypes>... validatorParameter)
+      requires(sizeof...(validatorParameter) > 0 &&
+               sizeof...(ValidatorParameterTypes) ==
+                   sizeof...(validatorParameter)) {
+    addValidatorImpl(
+        "addValidator",
+        []<typename T>(ConstConfigOptionProxy<T> opt) {
+          return opt.getConfigOption().template getValue<std::decay_t<T>>();
+        },
+        validatorFunction, errorMessage,
+        static_cast<ConstConfigOptionProxy<ValidatorParameterTypes>>(
+            validatorParameter)...);
+  }
+
+  /*
+  @brief Add a function to the configuration manager for verifying, that the
+  given configuration options are valid. Will always be called after parsing.
+
+  @tparam ValidatorParameterTypes Must be instances of `ConfigOptionProxy`. Can
+  be left up to type deduction.
+
+  @param validatorFunction Checks, if the given configuration options are valid.
+  Should return true, if they are valid.
+  @param errorMessage A `std::runtime_error` with this as an error message will
+  get thrown, if the `validatorFunction` returns false.
+  @param validatorParameter Proxies for the configuration options, who will be
+  passed to the validator function as function arguments. Will keep the same
+  order.
+  */
+  template <isInstantiation<ConfigOptionProxy>... ValidatorParameterTypes>
+  void addOptionValidator(
+      Validator<decltype(std::declval<ValidatorParameterTypes>()
+                             .getConfigOption())...> auto validatorFunction,
+      std::string_view errorMessage,
+      ValidatorParameterTypes... validatorParameter)
+      requires(sizeof...(validatorParameter) > 0) {
```

Comment:
We have the impIf you have too much `decltype and declval` then I suggest trailing requires clauses:

```
template <typename ValidatorT>
void addVal(ValidatorT validator, isInstantiatcion<ConfigOptionProxy> auto... options) requires Validator<decltype(options.getConfigOption())...> {...}

Reply:
And this function (as well as the function above) should also work for `ConstConfigOptionProxy`.

Reply:
> We have the impIf you have too much `decltype and declval` then I suggest trailing requires clauses:
> 
> ```
> template <typename ValidatorT>
> void addVal(ValidatorT validator, isInstantiatcion<ConfigOptionProxy> auto... options) requires Validator<decltype(options.getConfigOption())...> {...}
> ```

I'm not 100% sure, but I think there is at least one sentence missing. I'm going to wait, until that has been fixed, before implementing changes.

Reply:
> And this function (as well as the function above) should also work for `ConstConfigOptionProxy`.

I'm not using `ConstConfigOptionProxy`, because I'm trying to keep it user friendly.

Alright, this needs a longer explanation: The return type of `ConfigManager::addOption` is `ConfigOptionProxy`, because `ConfigOption::addValidator` is a non const function.
Because the `ConfigManager::addValidator` and `ConfigManager::addOptionValidator` both are template functions, an implicit conversion from `ConfigOptionProxy` to `ConstConfigOptionProxy` is not possible.
And because I don't want the user to have to manually cast `ConfigOptionProxy` to `ConstConfigOptionProxy`, every time they add a validator of any kind, I simply use `ConfigOptionProxy`.

Understand the problem with that idea?

Reply:
All done in [96dc359](https://github.com/ad-freiburg/qlever/pull/1058/commits/96dc35935bcc33f7c0a8e9974f14b7c19ffd8a6e) and [084c01d](https://github.com/ad-freiburg/qlever/pull/1058/commits/084c01dce3d198509ad11bc109abb1d363fb36bb).